### ISSUE TITLE
Some questions about the code for FM and SAA module

### ISSUE THREAD

**Issue by zyfone** [a-shi321/US-DAF#1]
Hi, thanks for the open-source code.
But I have some questions to confirm

The code is not adopting the FM and SAA module in the global-level feature.
       

        FM
        base_score = self.RCNN_imageDA(base_feat)
        DA_img_loss_cls = nn.BCELoss()(base_score.view(-1, 1), torch.ones_like(base_score).view(-1, 1))

**Comment by a-shi321** [comment]
Since the instance level has a correspondence with the image level, we will map the weights obtained from the image-level to the image-level through the mapping relationship, as follows:
for i in range(len(gt_boxes2)):
                x1 = torch.tensor(math.floor(gt_boxes2[i][1] * scale_w))
                y1 = torch.tensor(math.floor(gt_boxes2[i][2] * scale_h))
                x2 = torch.tensor(math.floor(gt_boxes2[i][3] * scale_w))
                y2 = torch.tensor(math.floor(gt_boxes2[i][4] * scale_h))
                weight_gt = torch.ones((self.n_classes, y2 - y1 + 1, x2 - x1 + 1)).cuda() * cls_prob2[i][:].view( self.n_classes, 1, 1)

                label_img[:, y1:y2 + 1, x1:x2 + 1] = weight_gt

        image_softmax = label_img.view(self.n_classes, -1).t()
 

and

 for i in range(len(gt_tgt_rois2)):

                    t_x1 = torch.tensor(math.floor(gt_tgt_rois2[i][1] * t_scale_w))
                    t_y1 = torch.tensor(math.floor(gt_tgt_rois2[i][2] * t_scale_h))
                    t_x2 = torch.tensor(math.floor(gt_tgt_rois2[i][3] * t_scale_w))
                    t_y2 = torch.tensor(math.floor(gt_tgt_rois2[i][4] * t_scale_h))
                    tgt_weight_gt = torch.ones((self.n_classes,t_y2 - t_y1 + 1, t_x2 - t_x1 + 1)).cuda() * tgt_cls_prob[i][:].view( self.n_classes, 1, 1)

                    tgt_label_img[:, t_y1:t_y2 + 1, t_x1:t_x2 + 1] = tgt_weight_gt
        tgt_image_softmax = tgt_label_img.view(self.n_classes, -1).t()

**Comment by zyfone** [comment]
Thanks for your reply.
By rhe way, could you please share the paper link for Conflict-Alleviated Gradient Descent for Adaptive Object Detection.

**Comment by a-shi321** [comment]
The preprint version of the paper will be posted on arXiv soon. Thank you for your attention.
